### PR TITLE
AST-2150 - In megamenu, margin is added on Responsive devices.

### PR DESCRIPTION
**Pull request by GauravKhupse**
### Description

If we added top offset for the menu, this will conflict with megamenu on responsive devices
import this file:- [astra-settings-export-08-25-2022](https://share.bsf.io/kpuWkmmE)

### Screenshots

https://share.bsf.io/yAuQPLLn

### Types of changes

Bug fix (non-breaking change which fixes an issue) 

### How has this been tested?

Added in the card

### Checklist:
- [ ] My code is tested
- [ ] My code passes the PHPCS tests
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: inc/builder/type/header/menu/dynamic-css/dynamic.css.php [hunk 1]
```diff
@@ -235,6 +234,9 @@ function astra_hb_menu_dynamic_css( $dynamic_css, $dynamic_css_filtered = '' ) {
 				'margin-left'   => astra_responsive_spacing( $margin, 'left', 'tablet' ),
 				'margin-right'  => astra_responsive_spacing( $margin, 'right', 'tablet' ),
 			),
+			$selector . ' .main-header-menu > .menu-item > .sub-menu, ' . $selector . ' .main-header-menu > .menu-item > .astra-full-megamenu-wrapper' => array(
```
Comment (imnavanath) [inline inc/builder/type/header/menu/dynamic-css/dynamic.css.php:237]: @GauravKhupse `.sub-menu` selector is needed here? Please make sure this is not creating any conflict with theme's top offset / any space.

File: inc/builder/type/header/menu/dynamic-css/dynamic.css.php [hunk 2]
```diff
@@ -235,6 +234,9 @@ function astra_hb_menu_dynamic_css( $dynamic_css, $dynamic_css_filtered = '' ) {
...
+				'margin-top' => astra_get_css_value( '0', 'px' ),
```
Comment (imnavanath) [inline inc/builder/type/header/menu/dynamic-css/dynamic.css.php:238]:
```suggestion
				'margin-top' => '0',
```

File: inc/builder/type/header/menu/dynamic-css/dynamic.css.php [hunk 3]
```diff
@@ -281,6 +283,9 @@ function astra_hb_menu_dynamic_css( $dynamic_css, $dynamic_css_filtered = '' ) {
 				'margin-left'   => astra_responsive_spacing( $margin, 'left', 'mobile' ),
 				'margin-right'  => astra_responsive_spacing( $margin, 'right', 'mobile' ),
 			),
+			$selector . ' .main-header-menu > .menu-item > .sub-menu, ' . $selector . ' .main-header-menu > .menu-item > .astra-full-megamenu-wrapper' => array(
+				'margin-top' => astra_get_css_value( '0', 'px' ),
```
Comment (imnavanath) [inline inc/builder/type/header/menu/dynamic-css/dynamic.css.php:287]:
```suggestion
				'margin-top' => '0',
```

File: inc/builder/type/header/menu/dynamic-css/dynamic.css.php (hunk 1 above)
Comment (GauravKhupse) [reply inc/builder/type/header/menu/dynamic-css/dynamic.css.php:237]: It's working fine with the normal menu and offset:- https://share.bsf.io/Qwud5715

File: inc/builder/type/header/menu/dynamic-css/dynamic.css.php (hunk 3 above)
Comment (GauravKhupse) [reply inc/builder/type/header/menu/dynamic-css/dynamic.css.php]: Updated changes!

File: inc/builder/type/header/menu/dynamic-css/dynamic.css.php (hunk 2 above)
Comment (GauravKhupse) [reply inc/builder/type/header/menu/dynamic-css/dynamic.css.php]: Updated changes!

**Review by imnavanath — APPROVED** [review]

**Pull request merged**